crypto/keyring/record.go

```diff
@@ -28,45 +26,15 @@ func NewRecord(name string, pk cryptotypes.PubKey, item isRecord_Item) (*Record,
 
 // bz, err := cdc.Marshal(privKey) yields an error that's why I cast privKey to curve PrivKey to serialize it
 func NewLocalRecord(cdc codec.Codec, privKey cryptotypes.PrivKey) (*Record_Local, error) {
-
-	var (
-		err error
-		bz  []byte
-	)
-
-	privKeyType := privKey.Type()
-
-	// TODO fix that cdc.Marshal(privKey)
-	// ERROR cannot use privKey (variable of type "github.com/cosmos/cosmos-sdk/crypto/types".PrivKey) as 
-	//codec.ProtoMarshaler value in argument to cdc.Marshal: missing method Marshal
-	
-
-	switch privKeyType {
-	case "secp256k1":
-		priv, ok := privKey.(*secp256k1.PrivKey)
-		if !ok {
-			return nil, fmt.Errorf("unable to cast privKey to *secp256k1.PrivKey")
-		}
-
-		// TODO double check if we have to marshal priv as *secp256k1.PrivKey
-		bz, err = cdc.Marshal(priv)
-		if err != nil {
-			return nil, err
-		}
-
-	case "ed25519":
-		priv, ok := privKey.(*ed25519.PrivKey)
-		if !ok {
-			return nil, fmt.Errorf("unable to cast privKey to *ed25519.PrivKey")
-		}
-
-		bz, err = cdc.Marshal(priv)
-		if err != nil {
-			return nil, err
-		}
+	fmt.Println("NewLocalRecord privKey", privKey)
+	bz, err := cdc.MarshalInterface(privKey)
```

Comment:
Unable to resolve panic  caused by `cdc.MarshalInterface(privKey)` in [Test_runAddCmdBasic](https://github.com/cosmos/cosmos-sdk/blob/kerying-migration/client/keys/add_test.go#L22). The full test output can be found [here](https://play.golang.org/p/Mi9pr2fTaIW). As you can see in output, lines 34-35 have not been printed out. Would be grateful for some tips to fix it.

Reply:
Resolved!